src/EditorFeatures/Core/Implementation/Classification/SemanticClassificationUtilities.cs

```diff
@@ -149,5 +165,57 @@ internal static class SemanticClassificationUtilities
                 throw ExceptionUtilities.Unreachable;
             }
         }
+
+        private static async Task AddSemanticClassificationsAsync(
+            Document document,
+            TextSpan textSpan,
+            IClassificationService classificationService,
+            List<ClassifiedSpan> classifiedSpans,
+            CancellationToken cancellationToken)
+        {
+            var workspace = document.Project.Solution.Workspace;
+            var fullyLoadedStateTask = s_workspaceToFullyLoadedStateTask.GetValue(
+                workspace, w =>
+                {
+                    var workspaceLoadedService = workspace.Services.GetRequiredService<IWorkspaceStatusService>();
+                    return workspaceLoadedService.WaitUntilFullyLoadedAsync(CancellationToken.None);
+                });
+
+            // If we're not fully loaded try to read from the cache instead so that classifications appear up to date.
+            // New code will not be semantically classified, but will eventually when the project fully loads.
```

Comment:
❗ Roslyn should not be performing I/O prior to full solution load. `IsFullLoadedAsync` represents a period of high I/O that is reserved for other systems.

Reply:
Tagging @jinujoseph here. The point of the caching work is precisely to say that during load languages can use a bit of io to provide a nicer experience by loading the last computed value for a file (either from a previous session, or pulled down from the cloud). This inherently needs io as part of the core design as that's what a cache is.

Reply:
Summary of offline discussion: we do have an ongoing commitment to avoid interfering high priority solution load operations. Prior to enabling this feature for default scenarios, we should obtain concrete metrics for the expected cost of running this code path during solution load, in particular the disk I/O impact.

Separately from the cross-team investigation, we should measure the size of the database storage required to support this feature, as well as the cost of storing this data in response to an average edit.

Reply:
Data collected from investigation:

1. no additional IOs caused by this feature during solution load.
2. we do measure (unrelated to this feature) around 150k reads and 750us of IO in OOP during this time.  This is due to other features waking up and calling to OOP.  Note: tehse numbers are tiny and very unlikely to impact solution load substantively.
3. Memory usage of this cache is minimal.  About a 30:1 scale of source file to cache size.  This is also 4x smaller than the FAR cache.
4. We only compute the cache values once the solution is loaded.  So computation doesn't impact solution load.
5. We only compute the cache values for open documents.  So computation should not be overly taxing, and is much more in line with normal things like diagnostics when FSA is off.

Based on this, i believe there should be no perf blockers for going this route.  @sharwell anything else you would like more data on?

Reply:
Given the data we measured, i'm resolving this as acceptable absent new concerns.